core/src/main/java/org/infinispan/commands/write/RemoveCommand.java

```diff
@@ -103,7 +103,8 @@ public Object perform(InvocationContext ctx) throws Throwable {
    }
 
    public void notify(InvocationContext ctx, Object removedValue, Metadata removedMetadata, boolean isPre) {
-      notifier.notifyCacheEntryRemoved(key, removedValue, removedMetadata, isPre, ctx, this);
+      if (removedValue != null)
```

Comment:
@wburns You should see this

Reply:
That is fine with me. Unfortunately it has been this way forever, and I have always wanted to change it, just never had a specific reason at the time.